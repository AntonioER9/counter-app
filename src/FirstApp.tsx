
interface Props {
  title: string;
  subTitle?: string;
  name?: string;
}

export const FirstApp = ({ title, subTitle, name }: Props) => {

  // console.log(props);
  
  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}