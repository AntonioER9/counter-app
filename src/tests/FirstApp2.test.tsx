import { render, screen } from '@testing-library/react';
import { FirstApp } from '../FirstApp';


describe('Tests in <FirstApp />', () => {

    const title    = 'Hi, I am Antonio';
    const subTitle = 'I am subtitle';
    
    test('Should be must match the snapshot', () => {
        
        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();

    });

    test('Should be show the message "Hi, I am Antonio"', () => {
        
        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();
        // screen.debug();
    });

    test('Should be show the title in an h1', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    });

    test('Should be show the subtitle sent by props', () => {
        
        render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            />  
        );

        expect( screen.getAllByText(subTitle).length ).toBe(2);

    });


});