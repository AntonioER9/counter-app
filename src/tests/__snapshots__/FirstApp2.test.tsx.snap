// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> Should be must match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hi, I am Antonio
     
  </h1>
  <p />
  <p />
  <p />
</div>
`;
